cli/js/web/url.ts

```diff
@@ -282,7 +287,9 @@ export class URLImpl implements URL {
 
   set hostname(value: string) {
     value = String(value);
-    parts.get(this)!.hostname = encodeURIComponent(value);
+    try {
+      parts.get(this)!.hostname = encoding.encodeHostname(value);
+    } catch {}
```

Comment:
May throw something here? as in:
https://github.com/denoland/deno/blob/4155844a1bea35b3cf253efda717ee677151ea8a/cli/js/web/url.ts#L460

Reply:
Browsers don't do that.